docker-build/Dockerfile

```diff
@@ -0,0 +1,106 @@
+FROM openjdk:8u131-jdk
+
+LABEL version="0.1.0" \
+      description="Build n4js with Maven. Includes xvfb, node." \
+      maintainer="Continuous Delivery Team <staff-build-release@numberfour.eu>" \
+      eu.numberfour.dockerimage.name="numberfour/build-n4js" \
+      eu.numberfour.dockerimage.github="eclipse/n4js"
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+VOLUME /workspace
+WORKDIR /workspace
+
+ENTRYPOINT ["/usr/local/bin/mvn-entrypoint.sh"]
+CMD ["mvn"]
+
+#
+# Maven installation:
+# see https://github.com/carlossg/docker-maven/tree/ecf54b9839caed8aa2bcf9b8f7bb19594634ee89/jdk-8
+# and https://hub.docker.com/_/maven/
+#
+ARG MAVEN_VERSION=3.2.2
+ARG USER_HOME_DIR="/root"
+ARG SHA=cce5914cf8797671fc6e10c4e034b453d854edf711cbc664b478d0f04934cb76
+ARG BASE_URL=https://archive.apache.org/dist/maven/maven-3/${MAVEN_VERSION}/binaries
+
+RUN mkdir -p /usr/share/maven /usr/share/maven/ref && \
+    curl -fsSL -o /tmp/apache-maven.tar.gz ${BASE_URL}/apache-maven-$MAVEN_VERSION-bin.tar.gz && \
+    echo "${SHA}  /tmp/apache-maven.tar.gz" | sha256sum -c - && \
+    tar -xzf /tmp/apache-maven.tar.gz -C /usr/share/maven --strip-components=1 && \
+    rm -f /tmp/apache-maven.tar.gz && \
+    ln -s /usr/share/maven/bin/mvn /usr/bin/mvn
+
+ENV MAVEN_HOME   /usr/share/maven
+ENV MAVEN_CONFIG "$USER_HOME_DIR/.m2"
+ENV MAVEN_OPTS   "-Xms512m -Xmx2048M"
+
+COPY mvn-entrypoint.sh /usr/local/bin/mvn-entrypoint.sh
+COPY settings-docker.xml /usr/share/maven/ref/
+
+#
+# The remaining part has been inspired by:
+# https://github.com/nodejs/docker-node/blob/master/8.9/wheezy/Dockerfile
+#
+
+RUN groupadd --gid 1000 node && \
+    useradd --uid 1000 --gid node --shell /bin/bash --create-home node
+
+RUN set -ex && \
+    for key in \
+        6A010C5166006599AA17F08146C2130DFD2497F5 \
+        9554F04D7259F04124DE6B476D5A82AC7E37093B \
+        94AE36675C464D64BAFA68DD7434390BDBE9B9C5 \
+        FD3A5288F042B6850C66B31F09FE44734EB7990E \
+        71DCFD284A79C3B38668286BC97EC7A07EDE3FC1 \
+        DD8F2338BAE7501E3DD5AC78C273792F7D83545D \
+        B9AE9905FFD7803F25714661B63B535A4C206CA9 \
+        C4F0DFFF4E8C1A8236409D08E73BC641CC11F4C8 \
+        56730D5401028683275BD23C23EFEFE93C4CFFFE \
+        77984A986EBC2AA786BC0F66B01FBB92821C587A ; \
+    do \
+        gpg --keyserver pgp.mit.edu                --recv-keys "$key" || \
```

Comment:
I get timeouts. Seems to be related to https://unix.stackexchange.com/questions/75892/keyserver-timed-out-when-trying-to-add-a-gpg-public-key.

changing to
```
        gpg --keyserver hkp://pgp.mit.edu:80                --recv-keys "$key" || \
        gpg --keyserver hkp://keyserver.pgp.com:80          --recv-keys "$key" || \
        gpg --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-keys "$key" ; \
```
solved the problem